powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/service/util/SdkDataWriterTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2023 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.getlime.security.powerauth.app.server.service.util;
+
+import org.bouncycastle.util.encoders.Base64;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Test for SDK data writer.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+public class SdkDataWriterTest {
+
+    @Test
+    public void testCountWrite() {
+        final String countsGeneratedBySdk = "AAF/gICA/4EAgQG//8AAQADAAP//wAEAAMD////BAAAAwQIDBNAgMED/////";
+        final SdkDataReader readerSdk = new SdkDataReader(Base64.decode(countsGeneratedBySdk));
+        final List<Long> expectedCounts = new ArrayList<>();
+        expectedCounts.add(0L);
```

Comment:
I think that you can use `Integer` here. The mobile SDK is using `size_t` for "count" because its bit-length depends on the architecture. I would get a ton of warnings if I use `int` or `long` instead, because the C++ standard template library is using `size_t` for "size" or "count"

The maximum value for our serialized count never exceeds 32-bit integer, so we're OK with int.

Reply:
Ah yes, I will fix this.